src/main/scala/nl/knaw/dans/easy/fsrdb/FsRdbUpdater.scala

```diff
@@ -275,4 +218,16 @@ object FsRdbUpdater {
     statement.closeOnCompletion()
   }
 
+  private def deleteDatasetItems(conn: Connection, datasetSid: String): Try[Unit] = Try {
+    log.debug(s"Attempting to delete files and folders for dataset $datasetSid")
+    val deleteFilesStatement = conn.prepareStatement("DELETE FROM easy_files WHERE dataset_sid = ?")
```

Comment:
Without any validation of the datasetSid we introduce injection vulnerability. Preparing statements as done for insertFile and insertFolder would fix the vulnerability. findPids suffers from the same vulnerability but needs another solution.